src/index.ts

```diff
@@ -0,0 +1,161 @@
+      this.originUrl = CheckoutInstance.TEST_URL
+    }
+    this.messageHandler = this.handleMessage.bind(this)
+    window.addEventListener('message', this.messageHandler)
+    const iframe = document.createElement('iframe')
+    iframe.setAttribute('src', `${this.originUrl}?manifest=${this.encodeManifest(manifest)}`)
+    // Using the attributes below in order to provide defaults without overriding CSS styles
+    iframe.setAttribute('width', '400')
+    iframe.setAttribute('height', '700')
+    iframe.setAttribute('frameborder', '0')
+    document.getElementById(this.options.id!)?.appendChild(iframe)
+  }
+
+  /**
+   * Validates the necessary parameters for Checkout initialization and gives helpful messages for integrators.
+   */
+  private validateParameters(manifest: CheckoutManifest, options: CheckoutOptions) {
+    if (!manifest || typeof manifest.id !== 'string' || manifest.id === '' || typeof manifest.session !== 'string' || manifest.session === '') {
+      console.error(`${CheckoutInstance.LOGTAG} Please provide a valid Checkout Manifest when calling startCheckout.`)
+      return false
+    }
+    if (typeof options.id !== 'string' || options.id === '') {
+      console.error(`${CheckoutInstance.LOGTAG} The HTML element id must be a non-empty string.`)
+      return false
+    }
+    const el = document.getElementById(options.id!)
+    if (!el) {
+      console.error(`${CheckoutInstance.LOGTAG} Could not find element ${options.id}.`)
+      return false
+    }
+    if (typeof options.onMessage !== 'function') {
+      console.error(`${CheckoutInstance.LOGTAG} The onMessage callback must be a function.`)
+      return false
+    }
+    if (typeof options.testing !== 'boolean') {
+      console.error(`${CheckoutInstance.LOGTAG} The testing option must be true or false.`)
+      return false
+    }
+    return true
+  }
+
+  /**
+   * Encodes a Manifest for iframe consumption. Supports both Node and browser environments.
+   */
+  private encodeManifest(manifest: CheckoutManifest) {
+    if (typeof Buffer !== 'undefined') {
+      return Buffer.from(JSON.stringify(manifest), 'binary').toString('base64')
+    } else {
+      return window.btoa(JSON.stringify(manifest))
+    }
+  }
+
+  /**
+   * Handles messages sent from the Checkout iframe. If the origin and contents are as expected,
+   * pass them on to the message handler that was configured in startCheckout.
+   *
+   * If the Checkout becomes completed, automatically removes the event listener.
+   */
+  private handleMessage(e: MessageEvent) {
+    if (e.origin === this.originUrl && e.data.type === 'ep-checkout') {
+      this.options.onMessage!(e.data.status)
+      if (e.data.status === 'complete') {
+        if (this.messageHandler) {
+          window.removeEventListener('message', this.messageHandler)
+        }
+      }
+    }
+  }
+
+  /**
+   * Used to cleanup Checkout by removing the DOM contents and event listener.
+   */
+  remove() {
```

Comment:
Change this to `unmount`, since `remove` is not the best definition and word for the intended action. Stripe uses unmount too btw.